### PR TITLE
New Yaml Package module

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Adds a package module for doing IO with yaml files.

Contains a `YamlSerializer` and `YamlDeserializer` for writing and parsing yaml files via the IO module's (de)serializer framework. Also has a class-based representation of Yaml values which can be written out or parsed into using the `writeYamlFile` and `parseYamlFile` procedures respectively.

This module depends on [libyaml](https://pyyaml.org/wiki/LibYAML), which is an external C library for writing and parsing yaml files.

- [X] inspect built docs
- [X] new tests passing locally
- [x] new tests passing with `-memleaks`
- [x] new tests passing with `-valgrind`
- [x] install `libyaml` on a testing system
- [x] full paratest

### CODE REVIEW

File: modules/packages/Yaml.chpl [hunk 1]
```diff
@@ -0,0 +1,1935 @@
...
+
+  record R {
+    var a: int;
+    var b: string;
+  }
+
+  var myFile = open("r.yaml", iomode.cwr),
+      r1 = new R(1, "hello");
+
+  myFile.writer().withSerializer(new YamlSerializer()).write(r1);
```
Comment (benharsh) [inline modules/packages/Yaml.chpl:43]: FWIW you should be able to write ``myFile.writer(serializer=new YamlSerializer()).write(r1);`` instead.

**Review by benharsh — APPROVED** [review]
Awesome!

Eventually it would be nice to see some more tests for things like ``YamlValue``, but this can go in as-is.

**Comment by bradcray** [comment]
Nice!

**Comment by bradcray** [comment]
Belated naming question that I'm only realizing as I'm writing up the release announcement:  By our naming conventions, wouldn't `YAML` (all caps) be a better spelling of this module's name?

**Pull request merged**